examples/src/test/scala/docs/guide/ShoppingCartApp.scala

```diff
@@ -102,165 +99,88 @@ object ShoppingCartApp extends App {
-  def checkoutCountsForDate(date: LocalDate): Future[Seq[Checkout]] = {
-    session
-      .selectAll(s"SELECT cart_id, item_id, quantity FROM $keyspace.$dailyCheckoutsTable WHERE date = ?", date)
-      .map { rows =>
-        rows.map(row => Checkout(date, row.getString("cart_id"), row.getString("item_id"), row.getInt("quantity")))
-      }
-  }
+  val cartStateTable = "cart_checkout_state"
 
-  def addCheckout(checkoutItem: Checkout): Future[Done] = {
-    import checkoutItem._
+  def updateCart(cartId: String): Future[Done] = {
     session.executeWrite(
-      s"INSERT INTO $keyspace.$dailyCheckoutsTable (date, cart_id, item_id, quantity) VALUES (?, ?, ?, ?)",
-      date,
-      cartId,
-      itemId,
-      Int.box(quantity))
-  }
-
-  def cartState(): Future[Seq[CartState]] = {
-    session
-      .selectAll(s"SELECT cart_id, item_id, quantity FROM $keyspace.$cartStateTable")
-      .map { rows =>
-        rows.map(row => CartState(row.getString("cart_id"), row.getString("item_id"), row.getInt("quantity")))
-      }
+      s"UPDATE $keyspace.$cartStateTable SET last_updated = ? WHERE cart_id = ?",
+      Instant.now(),
+      cartId)
   }
 
-  def updateCartState(cartState: CartState): Future[Done] = {
-    import cartState._
+  def checkoutCart(checkout: Checkout): Future[Done] = {
     session.executeWrite(
-      s"UPDATE $keyspace.$cartStateTable SET quantity = ? WHERE cart_id = ? AND item_id = ?",
-      Integer.valueOf(quantity),
-      cartId,
-      itemId)
-  }
-
-  def deleteCartState(cartId: String): Future[Done] = {
-    session.executeWrite(s"DELETE FROM $keyspace.$cartStateTable WHERE cart_id = ?", cartId)
+      s"UPDATE $keyspace.$cartStateTable SET last_updated = ?, checkout_time = ? WHERE cart_id = ?",
+      Instant.now(),
+      checkout.checkoutTime,
+      checkout.cartId)
   }
 }
 //#guideProjectionRepo
 
 //#guideProjectionHandler
-object DailyCheckoutProjectionHandler {
-  final case class Checkout(date: LocalDate, cartId: String, itemId: String, quantity: Int)
-  final case class CartState(cartId: String, itemId: String, quantity: Int)
+object CheckoutProjectionHandler {
+  val CheckoutLogInterval = 10
+  val CheckoutFormat = "%-12s%-9s"
 
-  def toDate(date: Instant) = LocalDate.from(date.atZone(ZoneId.of("UTC")))
+  final case class Checkout(cartId: String, checkoutTime: Instant)
 }
 
-class DailyCheckoutProjectionHandler(tag: String, system: ActorSystem[_], repo: DailyCheckoutProjectionRepository)
-    extends StatefulHandler[Map[String, Map[String, CartState]], EventEnvelope[ShoppingCartEvents.Event]]()(
-      system.classicSystem.dispatcher) {
+class CheckoutProjectionHandler(tag: String, system: ActorSystem[_], repo: CheckoutProjectionRepository)
+    extends Handler[EventEnvelope[ShoppingCartEvents.Event]]() {
 
-  import DailyCheckoutProjectionHandler._
+  import CheckoutProjectionHandler._
 
-  @volatile private var checkoutCounter = 0
+  @volatile private var checkouts: immutable.Seq[Checkout] = Nil
```

Comment:
```suggestion
  private var checkouts: immutable.Seq[Checkout] = Nil
```

As documented here: https://doc.akka.io/docs/akka-projection/current/cassandra.html#stateful-handler

We don't need volatile vars in a handler.

Reply:
> We don't need volatile vars in a handler

Even if the `Handler` is stateless? I guess It's ok, because we invoke `process` on a `mapAsync(1)`